service/controller/clusterapi/v29/cluster_resource_set.go

```diff
@@ -91,13 +91,13 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*controller.Resourc
 		}
 	}
 
-	var detectionService *detection.Detection
+	var clusterDetection *detection.Cluster
```

Comment:
```suggestion
	var clusterChangeDetection *detection.ClusterChange
```

I'd like to have `Change` there as this is not about detecting clusters' presence as such but actual changes in them.

Reply:
I would than rather like to rename the package to `changedetection`. This makes the package scope more specific and keeps the internals simpler.

Reply:
That makes sense :+1: